invenio_rdm_records/services/schemas/parent/access.py

```diff
@@ -71,6 +71,9 @@ class AccessSettingsSchema(Schema):
     allow_guest_requests = fields.Boolean()
 
     accept_conditions_text = SanitizedHTML(allow_none=True)
+    secret_link_expiration = fields.Integer(
```

Comment:
The conversion should happen here :) You can use a [pre/post_load](https://marshmallow.readthedocs.io/en/stable/extending.html#pre-processing-and-post-processing-methods) to achieve that.